resources/providers/awslib/ec2/provider.go

```diff
@@ -231,3 +232,18 @@ func (p *Provider) DescribeSnapshots(ctx context.Context, snapshot EBSSnapshot)
 	}
 	return result, nil
 }
+
+func (p *Provider) DeleteSnapshots(ctx context.Context, ids []string) error {
+	_, err := awslib.MultiRegionFetch(ctx, p.clients, func(ctx context.Context, region string, c Client) ([]Ec2Instance, error) {
```

Comment:
Note that a snapshot resides in a single region, and so, deleting a snapshot should be done only in the region where it exists. In that case, you can accept an argument of type `EBSSnapshot` as it has the region and also the snapshot id. (see example in DescribeSnapshots)
Since we don't know if all the snapshots are of the same region, the provider can accept a single `EBSSnapshot`, and you could rename the method to `DeleteSnapshot` (singular).